server/controllers/automationController.js

```diff
@@ -18,11 +19,55 @@ export default class AutomationController {
    * @param {object} res REST Response object
    * @returns {object} Response containing status message and automation data
    */
+
   static getAutomations(req, res) {
```

Comment:
to fix the code climate issue with code to long try this:
```javascript
static async getAutomations(req, res) {
    const modelOptions = {
      include,
      order,
    };
    const response = {
      status: 'success',
      message: 'Successfully fetched automations',
    };
    let countData;
    // if url doesn't contain a query objects send back all the data
    if ((Object.entries(req.query).length !== 0 && req.query.constructor === Object) || req.query.limit !== -1) {
      countData = await automation.findAndCountAll();
      response.pagination = {};
      response.pagination.page = parseInt(req.query.page, 10) || 1; // current page number
      const limit = parseInt(req.query.limit, 10) || 10;
      const offset = limit * (response.pagination.page - 1);
      response.pagination.numberOfPages = Math.ceil(countData.count / limit); // all pages count

      // check if number of pages is less than the current page number to show next page number
      if (response.pagination.page < response.pagination.numberOfPages) {
        response.pagination.nextPage = response.pagination.page + 1;
      }
      // show previous page number if page is greater than 1
      if (response.pagination.page > 1) {
        response.pagination.prevPage = response.pagination.page - 1;
      }
      modelOptions.offset = offset;
      modelOptions.limit = limit;
    }
    const allData = await automation.findAll({ ...modelOptions });
    response.data = formatAutomationResponse(allData);
    res.status(200).json(response);
  }
```
you can still reduce it further by separating some part into private functions

Reply:
Thanks. Working on it.